config.sample

```diff
@@ -1,65 +1,107 @@
 <?php
+# is INCLUDED in many (all?) of the OpenCrisisOnline pages
+# opens databases as well as other functions
+# defines db as the database handle
 
 # Help line config
 $anonymous = FALSE;                  # Setting to FALSE will inform both caller and volunteer who they talked to
-$help_line_number = "+16505551212";  # Must include prefix of +1 followed by phone number, no spaces or other characters
```

Comment:
Why is all the sample data erased?  That is done to help the user quickly edit the file. 

When the new admin sees "+16505551212" they know exactly what format you they supposed to enter right away.  Forcing them to read the directions will at best slow them down, and at worst create a problem with a bad config.

Reply:
Differences in design philosophy. With assigned data, the "dear" user/implementer might accept these as defaults and the program would run though with unexpected results as +16505551212 is an appropriate formatted number but not the number that one might be expecting.

Tell ya what: go through and identify the sample data that I deleted and/or changed. I will then go through and test each of the variables against the "default" (properly formatted data though probably isn't the value the user wants). If the default is detected will echo an error for each of the defaults found. Once all the identified variables are tested and if an default value is found, the code will then exit. 

Good enough @dustball ? (just calling out on this one as it does need a reply from you)

Reply:
Hmm. Let me think more about this.

One can programmatically check to see if the values are proper, and that will solve the "bad data" issue, but it doesn't address the "taking out sample data makes it harder to understand what to put in the first place and will increase the amount of time it takes to setup and potentially create a hurdle to usage" issue.  

As of right now I feel like removing sample data will result in it taking notably longer to fill in the file.  It looks intimidating and confusing without it IMHO.  But again lemme think about it.

Reply:
Uhm, sorry. Example - leave in:

`$help_line_number = "+16505551212";  # Must include prefix of +1 followed by phone number, no spaces or other characters`

Then in the setup.php, have the code check for $help_line_number being equal to "+16505551212". If equal, give fatal error message that $help_line_number in config.php must be set to an appropriate number.

Hey, wait, the config.php is sorta ugly now. What the lleh am I supposed to do says the implementer. What about comment having each assignment's comment start with #TODO:: and then details about the assignment statement. My #TODO comments starting in the first column would be deleted and your comments would stay in place but prefaced with the #TODO:: construct

Reply:
I'm going to repeat what you said in my own words to make sure I understand -

You propose:

- we go back to including the sample data
- we add extra checks to make sure the data is formatted correctly
- preface the existing comments with #TODO:: 

Is my understanding correct?  Apologies if not.  

All sounds good to me.

That all seems fine and extra checks surely can only make it better.  I think the most important one to check may be the db_charset since that is a new variable config variable and won't be populated by anyone migrating (if there are any existing installations of OCL in the wild, I'd probably know about it, but just in case ..).

Reply:
Also, the joke about the password showing up as stars only works with sample data of `hunter2`... 🤣

Reply:
A code fragment should say it all: :-)

<?php

$help_line_number = "+16505551212";  #TODO:: Must include prefix of +1 followed by phone number, no spaces or other characters
$friendly_phone = "(650) 555-HELP";  #TODO:: Human friendly version of the above number

$error_text = NULL;
if ("+16505551212" == $help_line_number)
    $error_text = $error_text."Don't mooch a non-exisitent number if you really want to help people\n\n";
if ("(650) 555-HELP" == $friendly_phone)
    $error_text = $error_text."$friendly_phone is not friendly enough\n\n";

if (NULL != $error_text){
    echo "Errors found in configuration data:\n\n$error_text\n";
    curlUpAndDie("FML or Sucks To Be You");
}
Hope some humor was successfully sent. 

Your original comments will taken verbatim and suffixed with "TODO:: " to make for easier searching, etc. Hmm, wonder if PHP has doc strings similar to Python. If so, I'll do a docstring with TODO embedded.